packages/voyager-keycloak/src/AuthContextProvider.ts

```diff
@@ -13,7 +11,7 @@ export class KeycloakAuthContextProvider implements AuthContextProvider {
     this.authenticated = !!((req && req.kauth && req.kauth.grant) ? req.kauth.grant.access_token : undefined)
   }
 
-  public isAuthenticated () {
+  public isAuthenticated (): boolean {
```

Comment:
I haven't checked that before but this may be bug. Authenticated is basing on flag that is set once. User can be logged out etc. but authenticated will still return true.

Reply:
For the duration of a request the user is authenticated right? You can't log out in the middle of a request

Reply:
How it works is the AuthContextProvider is created on every incoming request. This `isAuthenticated` here is really just saying 'is there a valid token on the request?' Seeing as you cannot really log out in the middle of a request I don't think this is a bug. What are your thoughts on that?

Reply:
Ah.. I got impression that this is created once.